app/src/main/java/com/mlinde/marvelcomicsapp/MainActivity.kt

```diff
@@ -12,35 +19,29 @@ import dagger.hilt.android.AndroidEntryPoint
 @AndroidEntryPoint
 class MainActivity : AppCompatActivity() {
 
-    private lateinit var binding: ActivityMainBinding
-    private lateinit var viewPager: ViewPager
-    private lateinit var tabs: TabLayout
+        private lateinit var binding: ActivityMainBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        viewPager = binding.viewPager
-        tabs = binding.tabs
+        val navHostFragment =
+            supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
```

Comment:
You can find nav controller like that:
```
val navController =  Navigation.findNavController(this, R.id. nav_host_fragment)
```
But it's only my suggestion. You don't need to change it. It's ok.